### ISSUE TITLE
Cannot install QGIS 2.14.6

### ISSUE THREAD

**Issue by ReneMalenfant** [OSGeo/homebrew-osgeo4mac#177]
As of Oct. 13, 2016, QGIS-214 dies upon launch and will not install on MacOS 10.12.  Yet again a Qt issue.  Here are the error messages from installation: 
> Last 15 lines from /Users/rene/Library/Logs/Homebrew/qgis-214/01.cmake:
> -- Looking for gaiaDropTable in /usr/local/lib/libspatialite.dylib
> -- Looking for gaiaDropTable in /usr/local/lib/libspatialite.dylib - found
> -- Looking for gaiaStatisticsInvalidate in /usr/local/lib/libspatialite.dylib
> -- Looking for gaiaStatisticsInvalidate in /usr/local/lib/libspatialite.dylib - found
> -- Looking for spatialite_init_ex in /usr/local/lib/libspatialite.dylib
> -- Looking for spatialite_init_ex in /usr/local/lib/libspatialite.dylib - found
> CMake Error at /usr/local/Cellar/cmake/3.6.2/share/cmake/Modules/FindQt4.cmake:1328 (message):
>   Found unsuitable Qt version "5.6.2" from /usr/local/opt/qt5/bin/qmake, this code requires Qt 4.x
> Call Stack (most recent call first):
>   CMakeLists.txt:277 (FIND_PACKAGE)
> 
> 
> -- Configuring incomplete, errors occurred!

**Comment by ecerrillo** [comment]
Same here! 

The problem appears to be due to an upgrade of `qt5`.

**Comment by ecerrillo** [comment]
I noticed that a new formula, `qt-4`, appeared in osgeo/osgeo4mac tap. I installed it, but the problem persists. I have tried to reinstall `qgis-214` using both `--HEAD` and `--build-from-source` options.

**Comment by dakcarto** [comment]
@ReneMalenfant and @ecerrillo please, instead, try the `qgis2-ltr` formula (QGIS 2.14.7 LTR), which is built upon the specialized `qt-4` and associated formulae. The `qgis2-*` formulae (another being worked on) are meant to build against Qt4, which is actively being deprecated from Homebrew-core and it formulae.

Note: These formulae are all currently in flux, and I have not yet had the time to test them on macOS Sierra. My dev box is 10.11.6. Also, there are no bottles being built while this tap is being prepared for migration to Homebrew-geospatial; so, sorry, you will need to compile `qt-4`, et al.

**Comment by ReneMalenfant** [comment]
Thanks dakcarto.  I tried installing the ```qgis2-ltr``` release last night.  It's a no-go because there are still dependencies on homebrew-core's ```qt4```, which doesn't work on Sierra.

Specifically:

1. ```qscintilla2-qt4``` will not install (2.9.3_1 has the dependencies changed to osgeo's ```pyqt-qt4```) but the following error message still persists:

> Last 15 lines from /Users/rene/Library/Logs/Homebrew/qscintilla2-qt4/04.python:
> 
> python
> configure.py
> -o
> /usr/local/Cellar/qscintilla2-qt4/2.9.3_1/lib
> -n
> /usr/local/Cellar/qscintilla2-qt4/2.9.3_1/include
> --apidir=/usr/local/Cellar/qscintilla2-qt4/2.9.3_1/qsci
> --destdir=/usr/local/Cellar/qscintilla2-qt4/2.9.3_1/lib/qt-4/python2.7/site-packages/PyQt4
> --stubsdir=/usr/local/Cellar/qscintilla2-qt4/2.9.3_1/lib/qt-4/python2.7/site-packages/PyQt4
> --qsci-sipdir=/usr/local/Cellar/qscintilla2-qt4/2.9.3_1/share/sip
> --pyqt-sipdir=/usr/local/opt/pyqt-qt4/share/sip
> --spec=unsupported/macx-clang-libc++
> 
> Error: Unable to import PyQt4.QtCore. Make sure PyQt4 is installed.
> 

2. QGIS dependency ```gpsbabel``` requires homebrew-core's ```qt``.

There may be other broken dependencies, but these are the ones that pop out.

**Comment by dakcarto** [comment]
Hi @ReneMalenfant, thanks for the test compilation and report!

I've moved the `gpsbabel` dep to optional (nice catch). `qgis-ltr` has been updated.

> qscintilla2-qt4 will not install (even though 2.9.3_1 has the dependencies changed to osgeo's pyqt-qt4) and the following error message still persists. (Note: I have osgeo/osgeo4mac/pyqt-qt4 and homebrew-core's pyqt5 installed; I can't install core's pyqt because it depends on qt, which won't install on Sierra.)

Does the following contain PyQt4 symlinks?

`/usr/local/lib/qt-4/python2.7/site-packages/PyQt4`

Is your result of doing the following in Terminal similar to mine on 10.11.6?:
```
$> export PYTHONPATH=/usr/local/lib/qt-4/python2.7/site-packages
$> python
Python 2.7.12 (default, Oct 11 2016, 05:24:00) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import PyQt4.QtCore
>>> import sys
>>> sys.path
['', '/usr/local/Cellar/matplotlib/1.5.1/libexec/lib/python2.7/site-packages',
'/usr/local/lib/python2.7/site-packages/dotenv-0.0.5-py2.7.egg',
'/usr/local/lib/qt-4/python2.7/site-packages',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python27.zip',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old',
'/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload',
'/Users/me/Library/Python/2.7/lib/python/site-packages',
'/usr/local/lib/python2.7/site-packages',
'/usr/local/Cellar/numpy/1.11.2/libexec/nose/lib/python2.7/site-packages',
'/usr/local/lib/python2.7/site-packages/wx-3.0-osx_cocoa']
>>> 
```

**Comment by ReneMalenfant** [comment]
Hi, dakcarto.

Directory ```/usr/local/lib/qt-4/``` does not exist.  (And I double-checked, and yes, ```osgeo/osgeo4mac/qt-4``` **is** installed.)

Here's the result of entering what you've suggested.
```
$> export PYTHONPATH=/usr/local/lib/qt-4/python2.7/site-packages
$> python
Python 2.7.12 (default, Oct 11 2016, 05:20:59)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import PyQt4.QtCore
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named QtCore
>>> import sys
>>> sys.path
['', '/usr/local/lib/python2.7/site-packages/HTSeq-0.6.1-py2.7-macosx-10.11-x86_64.egg', '/usr/local/lib/qt-4/python2.7/site-packages', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python27.zip', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old', '/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/site-packages', '/usr/local/Cellar/matplotlib/1.5.1/libexec/lib/python2.7/site-packages', '/usr/local/Cellar/numpy/1.11.2/libexec/nose/lib/python2.7/site-packages', '/usr/local/lib/python2.7/site-packages/wx-3.0-osx_cocoa', '/Library/Python/2.7/site-packages/pip-6.0.8-py2.7.egg', '/Library/Python/2.7/site-packages']
>>>
```

**Comment by dakcarto** [comment]
Hmm.

Try the following:
```
brew postinstall qt-4 qwt-qt4 qwtpolar-qt4 sip-qt4 pyqt-qt4 qscintilla2-qt4
```

Is the `/usr/local/lib/qt-4/` dir _still_ missing? e.g. (for me, using `tree`):
```
$> tree /usr/local/lib/qt-4/
/usr/local/lib/qt-4/
├── plugins
│   └── designer
│       ├── libqgis_customwidgets.2.14.7.dylib -> ../../../../opt/qgis2-ltr/lib/qt-4/plugins/designer/libqgis_customwidgets.2.14.7.dylib
│       ├── libqgis_customwidgets.2.16.3.dylib -> ../../../../opt/qgis2/lib/qt-4/plugins/designer/libqgis_customwidgets.2.16.3.dylib
│       ├── libqscintillaplugin.dylib -> ../../../../opt/qscintilla2-qt4/lib/qt-4/plugins/designer/libqscintillaplugin.dylib
│       ├── libqwt_designer_plugin.dylib -> ../../../../opt/qwt-qt4/lib/qt-4/plugins/designer/libqwt_designer_plugin.dylib
│       └── libqwt_polar_designer_plugin.dylib -> ../../../../opt/qwtpolar-qt4/lib/qt-4/plugins/designer/libqwt_polar_designer_plugin.dylib
└── python2.7
    └── site-packages
        ├── PyQt4
        │   ├── Qsci.pyi -> ../../../../../opt/qscintilla2-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qsci.pyi
        │   ├── Qsci.so -> ../../../../../opt/qscintilla2-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qsci.so
        │   ├── Qt.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qt.so
        │   ├── QtCore.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtCore.so
        │   ├── QtDeclarative.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtDeclarative.so
        │   ├── QtDesigner.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtDesigner.so
        │   ├── QtGui.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtGui.so
        │   ├── QtHelp.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtHelp.so
        │   ├── QtMultimedia.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtMultimedia.so
        │   ├── QtNetwork.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtNetwork.so
        │   ├── QtOpenGL.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtOpenGL.so
        │   ├── QtScript.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtScript.so
        │   ├── QtScriptTools.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtScriptTools.so
        │   ├── QtSql.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtSql.so
        │   ├── QtSvg.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtSvg.so
        │   ├── QtTest.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtTest.so
        │   ├── QtWebKit.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtWebKit.so
        │   ├── QtXml.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtXml.so
        │   ├── QtXmlPatterns.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtXmlPatterns.so
        │   ├── __init__.py -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/__init__.py
        │   ├── __init__.pyc
        │   ├── phonon.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/phonon.so
        │   ├── pyqtconfig.py -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/pyqtconfig.py
        │   ├── pyqtconfig.pyc
        │   └── uic -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/uic
        ├── sip.pyi -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sip.pyi
        ├── sip.so -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sip.so
        ├── sipconfig.py -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sipconfig.py
        ├── sipconfig.pyc
        └── sipdistutils.py -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sipdistutils.py
```

IMPORTANT NOTE: this is critical to the `qt-4`-based solution, because the `/usr/local/lib/qt-4` directory is necessary for both Qt4 plugins and Python modules, and is isolated from Homebrew-core installs

**Comment by dakcarto** [comment]
Note, the `libqgis_customwidgets*.dylib` are relative to `qgis2*` formula installs.

**Comment by ReneMalenfant** [comment]
Thanks, dakcarto.

That `postinstall` command seems to have worked, though including `qscintilla2-qt4` in it creates an error message because it was not already installed.  However, after running the `postinstall` command, I was successfully able to install `qscintilla2-qt4`.

`/usr/local/lib/qt-4/python2.7/site-packages/PyQt4` now contains the symlinks (although /usr/local/lib/python2.7/site-packages/PyQt4/Qsci.so is broken) , and `import PyQt4.QtCore` in python no longer causes an error.

Here is the tree result:
```
/usr/local/lib/qt-4/
├── plugins
│   └── designer
│       ├── libqscintillaplugin.dylib -> ../../../../opt/qscintilla2-qt4/lib/qt-4/plugins/designer/libqscintillaplugin.dylib
│       ├── libqwt_designer_plugin.dylib -> ../../../../opt/qwt-qt4/lib/qt-4/plugins/designer/libqwt_designer_plugin.dylib
│       └── libqwt_polar_designer_plugin.dylib -> ../../../../opt/qwtpolar-qt4/lib/qt-4/plugins/designer/libqwt_polar_designer_plugin.dylib
└── python2.7
    └── site-packages
        ├── PyQt4
        │   ├── Qsci.pyi -> ../../../../../opt/qscintilla2-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qsci.pyi
        │   ├── Qsci.so -> ../../../../../opt/qscintilla2-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qsci.so
        │   ├── Qt.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/Qt.so
        │   ├── QtCore.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtCore.so
        │   ├── QtDeclarative.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtDeclarative.so
        │   ├── QtDesigner.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtDesigner.so
        │   ├── QtGui.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtGui.so
        │   ├── QtHelp.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtHelp.so
        │   ├── QtMultimedia.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtMultimedia.so
        │   ├── QtNetwork.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtNetwork.so
        │   ├── QtOpenGL.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtOpenGL.so
        │   ├── QtScript.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtScript.so
        │   ├── QtScriptTools.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtScriptTools.so
        │   ├── QtSql.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtSql.so
        │   ├── QtSvg.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtSvg.so
        │   ├── QtTest.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtTest.so
        │   ├── QtWebKit.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtWebKit.so
        │   ├── QtXml.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtXml.so
        │   ├── QtXmlPatterns.so -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/QtXmlPatterns.so
        │   ├── __init__.py -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/__init__.py
        │   ├── __init__.pyc
        │   ├── pyqtconfig.py -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/pyqtconfig.py
        │   └── uic -> ../../../../../opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/uic
        ├── sip.pyi -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sip.pyi
        ├── sip.so -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sip.so
        ├── sipconfig.py -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sipconfig.py
        └── sipdistutils.py -> ../../../../opt/sip-qt4/lib/qt-4/python2.7/site-packages/sipdistutils.py

6 directories, 29 files
```

I am now installing `qgis2-ltr`; will report back when it is finished (or when it dies).

**Comment by ReneMalenfant** [comment]
Success!  Installed and the app opens.

Much appreciated.

**Comment by dakcarto** [comment]
Cool! Thanks for the feedback and testing @ReneMalenfant.

**Comment by dakcarto** [comment]
Final issue should be addressed with f8cb9bc.
